apps/web/src/pages/integrations/components/NovuInAppProviderModal.tsx

```diff
@@ -1,8 +1,170 @@
 import styled from '@emotion/styled/macro';
-import { colors, Text } from '../../../design-system';
+import { colors } from '../../../design-system';
+import { Accordion, Box, Center, Loader, useMantineTheme } from '@mantine/core';
 import { Close } from '../../../design-system/icons/actions/Close';
-import { Title, Center } from '@mantine/core';
-import { GotAQuestionButton } from '../../../components/utils/GotAQuestionButton';
+import { IIntegratedProvider } from '../IntegrationsStorePage';
+import { useEffect, useState } from 'react';
+import { createIntegration } from '../../../api/integration';
+import { useMutation } from '@tanstack/react-query';
+import { ChannelTypeEnum, ICredentialsDto } from '@novu/shared';
+import { useIntegrations } from '../../../hooks';
+import { SetupTimeline } from '../../quick-start/components/SetupTimeline';
+import { NovuInAppForm } from './NovuInAppForm';
+import { When } from '../../../components/utils/When';
+import { InAppSelectFramework } from './InAppSelectFramework';
+import { Faq } from '../../quick-start/components/QuickStartWrapper';
+import { SetupFrameworkHeader } from './SetupFrameworkHeader';
+import { FrameworkDisplay } from './FrameworkDisplay';
+
+export const NovuInAppProviderModal = ({
+  onClose,
+  provider: defaultProvider,
+  showModal,
+}: {
+  onClose: () => void;
+  provider: IIntegratedProvider | null;
+  showModal: (visible: boolean) => void;
+}) => {
+  const { refetch } = useIntegrations();
+  const [provider, setProvider] = useState<IIntegratedProvider | null>(defaultProvider);
+  const [isActive, setIsActive] = useState<boolean>(!!provider?.active);
+  const [framework, setFramework] = useState('');
+  const [page, setPage] = useState<'setup' | 'form' | 'framework'>('form');
+  const [created, setCreated] = useState(false);
+  const theme = useMantineTheme();
+
+  const { mutateAsync: createIntegrationApi, isLoading } = useMutation<
+    { _id: string; active: boolean },
+    { error: string; message: string; statusCode: number },
+    {
+      providerId: string;
+      channel: ChannelTypeEnum | null;
+      credentials: ICredentialsDto;
+      active: boolean;
+      check: boolean;
+    }
+  >(createIntegration, {
+    onSuccess: (data) => {
+      setCreated(true);
+      setPage('framework');
+      setProvider({
+        ...(provider as IIntegratedProvider),
+        integrationId: data._id,
+      });
+      setIsActive(data.active);
+      refetch();
+    },
+  });
+
+  useEffect(() => {
+    if (!provider || provider?.connected) {
+      return;
+    }
+    createIntegrationApi({
+      providerId: provider?.providerId ? provider?.providerId : '',
+      channel: provider?.channel ? provider?.channel : null,
+      credentials: {
+        hmac: false,
+      },
+      active: true,
+      check: false,
+    });
+  }, [provider?.connected]);
```

Comment:
IMO this should be called when the user is clicking on the "Novu In-app" card, not in the `useEffect`, then on success you can update the provider's `active` state.

Reply:
The first thing that happens when the user clicks on the card is that the modal is opened and this will be running.

Reply:
If this `connected` flag will change the effect might run twice, and I had this experience locally when was checking the code, you would get the error from the API.
It's always better to place such code in the callback to some actions, like user clicks.

Reply:
this is a potential source of bugs IMO... if we will switch in the future to React 18 we will get two requests there...